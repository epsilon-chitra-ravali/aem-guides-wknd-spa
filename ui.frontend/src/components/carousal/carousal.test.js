import React from 'react';
import HomeCarousel from './index';
import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom'

describe('DSTGridviewMicroFE test', () => {
  let renderedElem

  beforeEach(() => {
    renderedElem = render(<HomeCarousel />);
  });

  afterEach(() => {
    cleanup();
  })

  it('should match Snapshot', async () => {
    expect(renderedElem).toMatchSnapshot('s1');
  });

  it('should test on carousal click options', async () => {    
    const handleSelect = screen.getByLabelText("Slide 3"); 
    expect(handleSelect).toBeInTheDocument(); 
    fireEvent.click(handleSelect); 
    expect(renderedElem).toMatchSnapshot('s2');
  });
});