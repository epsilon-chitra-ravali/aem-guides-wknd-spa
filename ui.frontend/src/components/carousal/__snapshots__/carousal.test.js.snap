// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSTGridviewMicroFE test should match Snapshot: s1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel-parent"
      >
        <div
          class="carousel slide"
          data-testid="handle-select"
        >
          <div
            class="carousel-indicators"
          >
            <button
              aria-current="true"
              aria-label="Slide 1"
              class="active"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 2"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 3"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 4"
              data-bs-target=""
              type="button"
            />
          </div>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india1.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 1
                </h3>
                <p>
                  Description 1
                </p>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india2.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 2
                </h3>
                <p>
                  Description 2
                </p>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india3.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 3
                </h3>
                <p>
                  Description 3
                </p>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india4.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 4
                </h3>
                <p>
                  Description 4
                </p>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel-parent"
    >
      <div
        class="carousel slide"
        data-testid="handle-select"
      >
        <div
          class="carousel-indicators"
        >
          <button
            aria-current="true"
            aria-label="Slide 1"
            class="active"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 2"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 3"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 4"
            data-bs-target=""
            type="button"
          />
        </div>
        <div
          class="carousel-inner"
        >
          <div
            class="active carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india1.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 1
              </h3>
              <p>
                Description 1
              </p>
            </div>
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india2.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 2
              </h3>
              <p>
                Description 2
              </p>
            </div>
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india3.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 3
              </h3>
              <p>
                Description 3
              </p>
            </div>
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india4.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 4
              </h3>
              <p>
                Description 4
              </p>
            </div>
          </div>
        </div>
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DSTGridviewMicroFE test should test on carousal click options: s2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel-parent"
      >
        <div
          class="carousel slide"
          data-testid="handle-select"
        >
          <div
            class="carousel-indicators"
          >
            <button
              aria-current="false"
              aria-label="Slide 1"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 2"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="true"
              aria-label="Slide 3"
              class="active"
              data-bs-target=""
              type="button"
            />
            <button
              aria-current="false"
              aria-label="Slide 4"
              data-bs-target=""
              type="button"
            />
          </div>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item-start carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india1.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 1
                </h3>
                <p>
                  Description 1
                </p>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india2.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 2
                </h3>
                <p>
                  Description 2
                </p>
              </div>
            </div>
            <div
              class="carousel-item-next carousel-item-start carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india3.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 3
                </h3>
                <p>
                  Description 3
                </p>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slider image"
                class="d-block w-100"
                loading="lazy"
                src="Optimized-india4.webp"
                style="height: 40rem;"
              />
              <div
                class="carousel-caption"
              >
                <h3>
                  Caption 4
                </h3>
                <p>
                  Description 4
                </p>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel-parent"
    >
      <div
        class="carousel slide"
        data-testid="handle-select"
      >
        <div
          class="carousel-indicators"
        >
          <button
            aria-current="false"
            aria-label="Slide 1"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 2"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="true"
            aria-label="Slide 3"
            class="active"
            data-bs-target=""
            type="button"
          />
          <button
            aria-current="false"
            aria-label="Slide 4"
            data-bs-target=""
            type="button"
          />
        </div>
        <div
          class="carousel-inner"
        >
          <div
            class="active carousel-item-start carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india1.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 1
              </h3>
              <p>
                Description 1
              </p>
            </div>
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india2.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 2
              </h3>
              <p>
                Description 2
              </p>
            </div>
          </div>
          <div
            class="carousel-item-next carousel-item-start carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india3.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 3
              </h3>
              <p>
                Description 3
              </p>
            </div>
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="slider image"
              class="d-block w-100"
              loading="lazy"
              src="Optimized-india4.webp"
              style="height: 40rem;"
            />
            <div
              class="carousel-caption"
            >
              <h3>
                Caption 4
              </h3>
              <p>
                Description 4
              </p>
            </div>
          </div>
        </div>
        <a
          class="carousel-control-prev"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </a>
        <a
          class="carousel-control-next"
          href="#"
          role="button"
          tabindex="0"
        >
          <span
            aria-hidden="true"
            class="carousel-control-next-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
